src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,316 @@
+  const selfRef = ref || backupRef;
+  const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+  const [value, setValue] = useState(props.value || []);
+  const valueRefs = value.map(() => createRef());
+  const visibleOptions = getVisibleOptions();
+  const visibleOptionsRefs = visibleOptions.map(() => createRef());
+  const wrapperRef = useRef(null);
+
+  const ids = {
+    emptyMessage: `${props.id}-empty`,
+    label: `${props.id}-label`,
+    listbox: `${props.id}-listbox`,
+    textbox: `${props.id}-autocomplete`,
+  };
+
+  const classNames = {
+    container: styles.container,
+    formControlChildrenContainer: props.readOnly ? styles['form-control-children-container-readonly'] : styles['form-control-children-container'],
+    iconClear: props.readOnly ? styles['icon-clear-readonly'] : styles['icon-clear'],
+    iconsContainer: styles['icon-container'],
+    input: props.readOnly ? styles['input-readonly'] : styles.input,
+    noOptionsContainer: styles['no-options'],
+    popupContainer: styles['popup-container'],
+    tagList: styles['tag-list'],
+    ...props.classNames,
+  };
+
+  const dropdownContents = () => {
+    if (!expanded) {
+      return undefined;
+    }
+
+    if (props.showLoader) {
+      return (
+        <Listbox
+          className={classNames.popupContainer}
+          id={ids.listbox}
+          labelledBy={ids.label}
+          refs={[]}
+        >
+          <Loader inline />
+        </Listbox>
+      );
+    }
+
+    if (visibleOptions.length === 0) {
+      return (<NoOptions className={classNames.noOptionsContainer} id={ids.emptyMessage} />);
+    }
+
+    return (
+      <Listbox
+        className={classNames.popupContainer}
+        id={ids.listbox}
+        labelledBy={ids.label}
+        refs={visibleOptionsRefs}
+      >
+        {props.listboxChildren ?
+          props.listboxChildren(visibleOptions, visibleOptionsRefs, onOptionSelect) :
+          visibleOptions.map((option, index) => (
+            <ListOption
+              key={`${props.id}-${props.optionIDGetter(option)}`}
+              onSelect={onOptionSelect}
+              ref={visibleOptionsRefs[index]}
+              value={props.optionIDGetter(option)}
+            >
+              {props.optionLabelGetter(option)}
+            </ListOption>
+          ))}
+      </Listbox>
+    );
+  };
+
+  function getOption(id) {
+    return props.options.find(option => props.optionIDGetter(option) === id);
+  }
+
+  function getVisibleOptions() {
+    return props.options
+      .filter(option => props.optionLabelGetter(option).toLowerCase().includes(autocompleteValue.toLowerCase()))
+      .filter(option => !value.some((val) => {
+        const optionID = props.optionIDGetter(option);
+
+        if (typeof optionID === 'string') {
+          return val.toLowerCase() === optionID.toLowerCase();
+        }
+
+        return val === optionID;
+      }));
+  }
+
+  function onAutocompleteBlur() {
```

Comment:
Note: we're not using `useValidate` anymore, so this code is a one-off. We should refactor it after the appropriate changes to `useValidate` are made, or to use whatever replaces it.